packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -600,6 +613,9 @@ mod tests {
         Ok(())
     }
 
+    // TODO(oleksii): as expand_input_param function is gone, should all these
```

Comment:
Expanding function inputs now depends on the correct functioning of 

`Component::new`, `resolve_type` and `impl From<&ResolvedType> for TokenStream`.

A `Component` contains the function argument name (transformed to snake case if need be) and the `ResolvedType` of the argument. The behavior regarding the transformation of the argument name is covered by tests.

`resolve_type` works with the raw JSON until it results in a `ResolvedType` which, through `From<&ResolvedType> for TokenStream` can be converted into a valid Rust type signature -- e.g. `SomeStruct<u8>`. 

`resolved_type` should be unit tested to ensure we're resolving all the different types available (see ParamTypes). 

For some reason unknown to me, I have not written unit tests for `resolved_type` -- so its continued functioning depends on our e2e harness.

The `From<&ResolvedType> for TokenStream` should be tested as well.

So I'd focus my efforts there, either in this PR or another one.

Reply:
Added a new issue #635. Thanks!